src/bootsampling.jl

```diff
@@ -177,17 +177,21 @@ noise(bs::WildSampling) = bs.noise
 nvar(t::Tuple) = length(t)
 nvar(bs::BootstrapSample) = nvar(original(bs))
 
+tx(x::Tuple) = x
+tx(x::Number) = (x,)
 tx(x) = tuple(x...)
 
-## TODO: see Unroll.jl for a more efficient version, worth it?
-zeros_tuple(t, n) = tuple([zeros(typeof(x), n) for x in t]...)
+zeros_tuple(t::Tuple, m::Int) = _zeros_tuple(typeof(t), m)
+_zeros_tuple(::Type{Tuple{}}, m::Int) = ()
+_zeros_tuple(::Type{T}, m::Int) where T <: Tuple =
+    (zeros(Base.tuple_type_head(T), m), _zeros_tuple(Base.tuple_type_tail(T), m)...)
 
 lhs(f::Formula) = f.lhs
 
 """
 bootstrap(statistic, data, BasicSampling())
 """
-function bootstrap(statistic::Function, data, sampling::BasicSampling)
```

Comment:
Why do we need to drop `Function` here? Is it too restrictive?

Reply:
Yes, in my opinion it's too restrictive. It rules out the common use case of [function-like objects](https://docs.julialang.org/en/v1/manual/methods/#Function-like-objects-1).